enumerable.rb

```diff
@@ -0,0 +1,131 @@
+# module for some enumerable methods
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    new_array = []
+    i = 0
+    while i < size
+      new_array << yield(self[i])
+      i += 1
+    end
+    new_array
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    new_array2 = []
+    i = 0
+    while i < size
+      new_array2 << yield(self[i], i)
+      i += 1
+    end
+    new_array2
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    new_array3 = []
+    i = 0
+    while i < size
+      new_array3 << self[i] if yield(self[i]) == true
+      i += 1
+    end
+    new_array3
+  end
+
+  def my_all?
+    i = 0
+    while i < size
+      return false if yield(self[i]) == false || yield(self[i]).nil?
+
+      i += 1
+    end
+    true
+  end
```

Comment:
- [ ] when no block or argument is given returns true when none of the collection members are false or nil
- [ ] when a class is passed as an argument returns true if all of the collection is a member of such class
- [ ] when a Regex is passed as an argument returns true if all of the collection matches the Regex
- [ ] when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern